clap_derive/examples/arg_enum.rs

```diff
@@ -0,0 +1,19 @@
+use clap::Clap;
+
+#[derive(Clap, Debug, PartialEq)]
+enum ArgChoice {
+    Foo,
+    Bar,
+    Baz,
+}
+
+#[derive(Clap, PartialEq, Debug)]
+struct Opt {
+    #[clap(arg_enum)]
```

Comment:
I can actually make this `clap(enum)` but I would have to change `parse_clap_attribute` logic to check for `enum` token since it is a keyword

Reply:
So, `arg_enum` or `enum` finally?

Reply:
`arg_enum`. Or maybe `argument`. Pick the one you like most.